src/Mvc/Mvc.Core/test/Routing/UrlHelperTestBase.cs

```diff
@@ -288,6 +288,27 @@ public void IsLocalUrl_RejectsInvalidTokenUrls(string url)
             Assert.False(result);
         }
 
+        [Theory]
+        [InlineData("/\n")]
+        [InlineData("/\n/not-local-url")]
```

Comment:
Are control characters at other places allowed? I.e. at the end, in the middle.
If so, we need to check the whole url for control characters...

Reply:
AFAIK, `HttpRespose.Headers` will complain if it sees control characters anywhere a header value. We could choose to emulate it in this code. 

@Tratcher \ @blowdart what do you think?

Reply:
Return false or throw if it has any controls? Returning false would be the most compatible.

Reply:
Return false. But basically apply the same checks as whatever `HttpResponse.Headers` applies to determine the validity.

Reply:
Anything less than 0x20, or 0x7F (Delete)

Reply:
Is checking the ascii-range `[0x00, 0x7F]` enough?